test/e2e/driver.go

```diff
@@ -75,7 +75,9 @@ func RunE2ETests(authConfig, certDir, host, repoRoot, provider string, orderseed
 		r = append(r, reporters.NewJUnitReporter(path.Join(reportDir, "junit.xml")))
 	}
 	// Run the existing tests with output to console + JUnit for Jenkins
-	ginkgo.RunSpecsWithDefaultAndCustomReporters(&passed, "Kubernetes e2e Suite", r)
+	for i := 0; i < times && passed; i++ {
+		ginkgo.RunSpecsWithDefaultAndCustomReporters(&passed, fmt.Sprintf("Kubernetes e2e Suite run %d of %d", i+1, times), r)
+	}
```

Comment:
Fair point, it does look like the last one run will win. On the plus side, we stop doing new runs if we encounter an error, meaning that any reports that get overwritten will just be reports of successes.

I'd use a different reporter for each run (junit_%d.xml), but I imagine that existing consumers are dependent on the filename.